crates/solver/src/settlement_ranker.rs

```diff
@@ -182,6 +182,26 @@ impl SettlementRanker {
                 )),
             );
         }
+
+        // Even though some settlements passed the simulation, we require from solvers to have
+        // a bit more ETH balance then needed at this moment, to cover the potential increase of
+        // cost of sending transaction onchain, because of the sudden gas price increase
+        let rated_settlements = rated_settlements
+            .into_iter()
+            .filter(|(solver, settlement, _)| {
```

Comment:
Instead of calling `into_iter().filter().collect()` you could also call `rated_settlements.retain()`. Might look a little nicer.